api-cursors/src/main/java/org/zalando/nakadi/service/CursorsService.java

```diff
@@ -223,8 +225,19 @@ public void resetCursors(final String subscriptionId,
                 final List<SubscriptionCursorWithoutToken> newCursors = cursors.stream()
                         .map(cursorConverter::convertToNoToken)
                         .collect(Collectors.toList());
+                Predicate<Partition> isUpdatedPartition = p -> cursors.stream().anyMatch(c -> p.getKey().equals(c.getEventTypePartition()));
 
-                zkClient.closeSubscriptionStreams(() -> zkClient.forceCommitOffsets(newCursors), timeout);
+                zkClient.closeSubscriptionStreams(() -> {
+                    zkClient.forceCommitOffsets(newCursors);
+                    // reset the DLQ state of any partition that got updated
+                    zkClient.updateTopology(topology ->
+                        Arrays
+                            .stream(topology.getPartitions())
+                            .filter(isUpdatedPartition)
+                            .map(p -> p.toZeroFailedCommits().toLastDeadLetterOffset(null))
```

Comment:
```suggestion
                            .map(p -> p.toLastDeadLetterOffset(null))
```
I think that also resets the failed commits to 0?  Or should we have a function to do both things explicitly?